template/workflows/helm/.github/workflows/azure-kubernetes-service-helm.yml

```diff
@@ -131,5 +135,10 @@ jobs:
 
       - name: Deploy application on public cluster
         if: steps.isPrivate.outputs.PRIVATE_CLUSTER != 'true'
-        run: helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }} --create-namespace
+        run: |
+          if [ "${{ env.CLUSTER_RESOURCE_TYPE }}" == "Microsoft.ContainerService/fleets" ]; then
+            helm upgrade -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }}
+          else
+            helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }} --create-namespace
```

Comment:
Just a gentle reminder about #446, let's think about how we're going to parameterize `--create-namespace` here.  It's better to handle that in a new PR based on this branch as well.